### DISCUSSION TITLE
What is the intended way of setting `VCPKG_TARGET_TRIPLET`?

### DISCUSSION THREAD

**Discussion by dhrdlicka** [microsoft/vcpkg#22987] [Q&A]
My goal is to make sure the user can't choose a build configuration and target triplet combination that does not match (e.g. static build with a dynamic triplet, or x64 build with x86 libraries). Is there any proper way to set one and from that determine the other?

**Comment by autoantwort** [comment]
When using the vcpkg toolchain file `VCPKG_TARGET_TRIPLET` is automatically determined.

**Reply by dhrdlicka** [reply]
That's true, but it's not determined based on the project's settings. I am trying to see if there is a way I could set the target triplet based on how the user configures the project.

**Reply by autoantwort** [reply]
Is the project a library? What about adding it to vcpkg? Then you don't have this problem anymore.
